processor/src/lib.rs

```diff
@@ -0,0 +1,313 @@
+use std::collections::{BTreeMap, HashMap};
+use log::debug;
+use plonky2::field::goldilocks_field::GoldilocksField;
+use plonky2::field::types::Field;
+use vm_core::program::{Program, REGISTER_NUM};
+use vm_core::trace::instruction::{Add, Call, CJmp, Equal, ImmediateOrRegName, Instruction, Jmp, Mov, Mul, Opcode, Ret, Sub};
+use vm_core::trace::instruction::ImmediateOrRegName::Immediate;
+use crate::decode::{decode_raw_instruction, IMM_INSTRUCTION_LEN};
+use crate::memory::Memory;
+
+
+#[cfg(test)]
+mod tests;
+mod memory;
+mod decode;
+
+// r15 use as fp for procedure
+const FP_REG_INDEX: usize = 15;
+
+#[derive(Debug, Default)]
+pub struct Process {
+    pub clk: u32,
+    pub registers: [GoldilocksField; REGISTER_NUM],
+    pub pc: u64,
```

Comment:
```suggestion
    pub pc: u32,
```

Reply:
Match with GoldilocksField type. otherwise need many convert ugly.

Reply:
There is a method `from_canonical_u32` in GoldilocksField that can be used to convert u32 to Field.

Reply:
Another thing confuses me is that why`clk` u32 and `pc` u64?

Reply:
ahh, I can change clk to u64 next version